framework/auth/core.py

```diff
@@ -605,7 +659,8 @@ def add_unclaimed_record(self, node, referrer, given_name, email=None):
         record = {
             'name': given_name,
             'referrer_id': referrer_id,
-            'token': generate_confirm_token(),
+            'token': generate_verification_key(),
+            'expires': dt.datetime.utcnow() + dt.timedelta(days=settings.CLAIM_TOKEN_EXPIRATION),
```

Comment:
With suggested refactor [above](https://github.com/CenterForOpenScience/osf.io/pull/5964/files#r73884878), this could become `record.update(generate_verification_key(type=CLAIM))`